setup.go

```diff
@@ -0,0 +1,828 @@
+
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+type setupOptionsType struct {
+	configPath         string
+	deviceType         string
+	username           string
+	password           string
+	serverURL          string
+	serverIP           string
+	serverCert         string
+	tenantToken        string
+	invPollInterval    int
+	retryPollInterval  int
+	updatePollInterval int
+	hostedMender       bool
+	demo               bool
+}
+
+// ------------------------------ Setup constants ------------------------------
+const ( // state enum
+	stateDeviceType = iota
+	stateDeviceTypeConfirm
+	stateHostedMender
+	stateDemoMode
+	stateServerURL
+	stateServerIP
+	stateServerCert
+	stateHostedMenderCredentials
+	statePolling
+	stateDone
+)
+
+const ( // Prompt constants
+	promptWizard = `Mender Client Setup
+===================
+
+Setting up the Mender client: the client runs on your device as a daemon, and will regularly poll the server to check for updates and report its inventory data.
+
+Get started by first configuring the device type and settings for communicating with the server:
+`
+	promptDone = "Mender is setup successfully. You may now start the " +
+		"daemon by running the `daemon` command."
+	promptDoneRunDaemon = "Starting Mender client…\n" + // NOTE: format
+		"Your device should show up at %s within the next few " +
+		"minutes.\nLog in to your Mender UI at %s to accept it.\n"
+
+	promptDeviceType   = "Device type (for example raspberrypi3-raspbian): "
+	promptConfirmDev   = "Confirm devicetype as %s? [Y/n] " // NOTE: format
+	promptHostedMender = "\nAre you connecting this device to Hosted Mender? [Y/n] "
+	promptCredentials  = "\nEnter your Hosted Mender credentials"
+	promptDemoMode     = "\nDemo mode uses short poll intervals and assumes the " +
+		"default demo server setup. (Recommended for testing.)\n" +
+		"Do you want to run the client in demo mode? [Y/n] "
+
+	promptServerIP   = "\nSet the IP of the Mender Server: [127.0.0.1] "
+	promptServerURL  = "\nSet the URL of the Mender Server: [https://docker.mender.io] "
+	promptServerCert = "\nSet the location of the certificate of the server, " +
+		"if any (path to file, for example /etc/mender/server.crt): "
+	promptUpdatePoll = "\nSet the update poll interval - the frequency with " +
+		"which the client will send an update check request to the " +
+		"server, in seconds: [1800] "
+	promptRetryPoll = "\nThe retry poll interval defines the number of " +
+		"seconds between each attempt the client tries to communicate " +
+		"with the server. Note that the client may attempt more often " +
+		"initially to enable rapid upgrades, but will gradually fall " +
+		"back to this value if the server is busy.\n" +
+		"Set the retry poll interval [300]: "
+	promptInventoryPoll = "Set the inventory poll interval - the frequency " +
+		"with which the client will send inventory data to the server, " +
+		"in seconds: [28800] "
```

Comment:
General comment. Sorry I forgot in the previous round.

Most of the times that we ask for something that has a default, it will look like this:
- tool: Set some parameter [default value]
- me: type something wrong
- tool: Incorrect, please try again:
- me: mmm... what happens now if I press enter?

My point is that in the error/retry messages we should still have the default values printed... Do you agree? And what do you think @michaelatmender?